src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -1019,14 +1059,24 @@ public StreamDocumentActions(AsyncBlittableJsonTextWriter writer, DocumentsOpera
 
             public async ValueTask WriteDocumentAsync(DocumentItem item, SmugglerProgressBase.CountsWithLastEtagAndAttachments progress)
             {
-                if (item.Attachments != null)
-                    throw new NotSupportedException();
-
                 var document = item.Document;
                 using (document)
                 {
                     if (_options.OperateOnTypes.HasFlag(DatabaseItemType.Attachments))
-                        await WriteUniqueAttachmentStreamsAsync(document, progress);
+                    {
+                        if (item.Attachments != null)
+                        {
+                            foreach (var attachment in item.Attachments)
+                            {
+                                attachment.Stream.Position = 0;
+                                await WriteAttachmentStreamAsync(attachment.Base64Hash.Content.ToString(), attachment.Stream, attachment.Tag.ToString());
```

Comment:
check if we need this as `string` or can we keep it as `LazyStringValue`
same for the `Tag`

Reply:
We can't use LazyStringValue